airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockDestinationBackend.kt

```diff
@@ -21,6 +21,12 @@ object MockDestinationBackend {
         return getFile(filename)
     }
 
+    fun deleteOldRecords(filename: String, minGenerationId: Long) {
```

Comment:
do you think we can/should have a MockObjectStorage integration test? i.e. something to exercise your new toolkit, without actually relying on s3

or is s3 fast enough that that's not really necessary?

Reply:
Unit tests around the s3 interface, with an actual s3 for validation, makes more sense? Then separate unit tests around avro, csv, parquet conversion, using a mocked interface that dumps to a local file.